docs/api-version-4-migration-guide.md

```diff
@@ -0,0 +1,97 @@
+`SBOMFile` | `SbomFile`
+`SBOMFormatExtensions` | `SbomFormatExtensions`
+`SBOMMetadata` | `SbomMetadata`
+`SBOMPackagesProvider` | `SbomPackagesProvider`
+`SBOMReference` | `SbomReference`
+`SBOMRelationship` | `SbomRelationship`
+`SBOMTelemetry` | `SbomTelemetry`
+`SBOMValidationResult` | `SbomValidationResult`
+`SPDXSBOMReaderForExternalDocumentReference` | `SPDXSbomReaderForExternalDocumentReference`
+`ValidatedSBOM` | `ValidatedSbom`
+`ValidatedSBOMFactory` | `ValidatedSbomFactory`
+
+### Methods
+
+Old name (all caps) | New name (Pascal case)
+--- | ---
+`IInternalMetadataProvider.GetSBOMNamespaceUri` | `IInternalMetadataProvider.GetSbomNamespaceUri`
+`IRecorder.RecordSBOMFormat` | `IRecorder.RecordSbomFormat`
+`ISBOMGenerator.GetSupportedSBOMSpecifications` | `ISbomGenerator.GetSupportedSbomSpecifications`
+`ISBOMReaderForExternalDocumentReference.ParseSBOMFile`  | `ISbomReaderForExternalDocumentReference.ParseSbomFile`
+`ISbomRedactor.RedactSBOMAsync` | `ISbomRedactor.RedactSbomAsync`
+`ManifestInfo.ToSBOMSpecification` | `ManifestInfo.ToSbomSpecification`
+`SbomConfigProvider.GetSBOMNamespaceUri` | `SbomConfigProvider.GetSbomNamespaceUri`
+`SbomGenerator.GetSupportedSBOMSpecifications` | `SbomGenerator.GetSupportedSbomSpecifications`
+`SbomRedactor.RedactSBOMAsync` | `SbomRedactor.RedactSbomAsync`
+`SPDXSBOMReaderForExternalDocumentReference.ParseSBOMFile` | `SPDXSbomReaderForExternalDocumentReference.ParseSbomFile`
+`TelemetryRecorder.RecordSBOMFormat` | `TelemetryRecorder.RecordSbomFormat`
+`ValidatedSBOMFactory.CreateValidatedSBOM` | `ValidatedSbomFactory.CreateValidatedSbom`
+
+### Properties and const values
+
+Old name (all caps) | New name (Pascal case)
+--- | ---
+`Events.SBOMGenerationWorkflow` | `Events.SbomGenerationWorkflow`
+`Events.SBOMParseMetadata` | `Events.SbomParseMetadata`
+`Events.SBOMValidationWorkflow` | `Events.SbomValidationWorkflow`
+`FileHashes.SBOMFileHash` | `FileHashes.SbomFileHash`
+`IAssemblyConfig.DefaultSBOMNamespaceBaseUri` | `IAssemblyConfig.DefaultSbomNamespaceBaseUri`
+`MetadataKey.SBOMToolName` | `MetadataKey.SbomToolName`
+`MetadataKey.SBOMToolVersion` | `MetadataKey.SbomToolVersion`
+`SettingsSource.SBOMApi` | `SettingsSource.SbomApi`
+
+We discussed making a similar change with "SPDX", but ultimately decided to leave "SPDX" as all caps when dealing with interfaces and classes. There are many places in the code where variables use either camel casing or Pascal casing, but none of those should impact code that builds upon our API.
+
+## Functional changes
+
+If you use the Workflows that are documented [here](sbom-tool-cli-reference.md), then you will not be impacted by most of the functional changes. You will get the default implementations of the classes, which implement the current API. If, however, you create your own implementations or wrappers based on any of the interfaces exposed in our API, then you may need to make corresponding changes in your code. This table summarizes the changes:
```

Comment:
This link does not mention anything about workflows, should it be this instead? https://github.com/microsoft/sbom-tool/blob/main/docs/sbom-tool-api-reference.md

Reply:
Thanks! Corrected.